### ISSUE TITLE
FontFamily is subscribe but not unsubscribe

### ISSUE THREAD

**Issue by Coloryr** [AvaloniaUI/Avalonia#13703]
In window
```
<Window
    x:Class="ColorMC.Gui.UI.Windows.SingleWindow"
    xmlns="https://github.com/avaloniaui"
    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
    xmlns:d="http://schemas.microsoft.com/expression/blend/2008"
    xmlns:mc="http://schemas.openxmlformats.org/markup-compatibility/2006"
    xmlns:model="clr-namespace:ColorMC.Gui.UI.Model"
    xmlns:setting="clr-namespace:ColorMC.Gui.Utils.LaunchSetting"
    xmlns:views1="clr-namespace:ColorMC.Gui.UI.Windows"
    Title="Title"
    Width="760"
    Height="450"
    MinWidth="340"
    MinHeight="450"
    d:DesignHeight="450"
    d:DesignWidth="760"
    x:DataType="model:BaseModel"
    Background="{setting:Colors TranBack}"
    ExtendClientAreaChromeHints="NoChrome"
    ExtendClientAreaTitleBarHeightHint="-1"
    ExtendClientAreaToDecorationsHint="True"
    FontFamily="{setting:Font}"
    RequestedThemeVariant="{Binding Theme}"
    TransparencyBackgroundFallback="Transparent"
    TransparencyLevelHint="{Binding Hints}"
    mc:Ignorable="d">
    <views1:AllControl Name="Win" />
</Window>
```

FontFamily is subscribe but not unsubscribe

_Originally posted by @Coloryr in https://github.com/AvaloniaUI/Avalonia/discussions/13702#discussioncomment-7647794_

**Comment by Coloryr** [comment]
[AvaloniaApplication1.zip](https://github.com/AvaloniaUI/Avalonia/files/13445625/AvaloniaApplication1.zip)

**Comment by Coloryr** [comment]
![{D655863E-394D-4c8e-9C46-B28BFE9B3299}](https://github.com/AvaloniaUI/Avalonia/assets/26276037/af75536a-a44d-4d35-b14d-c749e1b990c0)
![{5FA5321E-300D-4a39-9EAD-DEB96991AC50}](https://github.com/AvaloniaUI/Avalonia/assets/26276037/352a9a41-2b26-4dbd-88e6-d884d843f315)
I think is not only `FontFamily`, it have other Property was not unsubscribe, and it have mem leak

**Comment by Coloryr** [comment]
use `ReflectionBindingExtension` without `IObserver.ToBinding()` is now mem leak

**Comment by maxkatz6** [comment]
Looks like a bug with observables. It never unsubscribes. ReflectionBinding uses weak references a lot, so it might not be a problem there.

For now, you can use weak references too:
```
public static class FontSel
{
    private static FontFamily s_font = new(FontFamily.DefaultFontFamilyName);
    private static readonly List<WeakReference<IObserver<FontFamily>>> s_fontList = [];

    public static IDisposable Add(IObserver<FontFamily> observer)
    {
        s_fontList.Add(new WeakReference<IObserver<FontFamily>>(observer));
        observer.OnNext(s_font);
        return new Unsubscribe(observer);
    }

    public static void Remove(IObserver<FontFamily> observer)
    {
        foreach (var item in s_fontList.ToArray())
        {
            if (!item.TryGetTarget(out var target)
                || target == observer)
            {
                s_fontList.Remove(item);
            }
        }
    }

    private class Unsubscribe(IObserver<FontFamily> observer) : IDisposable
    {
        public void Dispose()
        {
            Remove(observer);
        }
    }

    /// <summary>
    /// 刷新UI
    /// </summary>
    private static void Reload()
    {
        foreach (var item in s_fontList)
        {
            if (item.TryGetTarget(out var target))
            {
                target.OnNext(s_font);
            }
        }
    }
```

There shouldn't be memory leak this way.

**Comment by Coloryr** [comment]
Use `WeakReference` also have some leak, but not to much

**Comment by grokys** [comment]
If I understand correctly, in your repro (AvaloniaApplication1.zip) I should be repeatedly clicking the "NewWindow" button and closing the window. 

After doing that a few times, the problem is that a `AvaloniaApplication1.FontSel+Unsubscribe` object is leaking for each Window instance, is that correct?

To me the problem here is that if you bind with observables, you're responsible for cleaning up those bindings - in the general case, a control can't know when a binding should be disposed.

I mentioned in https://github.com/AvaloniaUI/Avalonia/pull/13816#issuecomment-1840482738 that I don't think special-casing `TopLevel` is the right thing to do here: sure it would solve this particular case, but what if you then use the `FontExtension` on a non-`TopLevel`? You again have a leak.

I think it might be useful to know what you're trying to do and find an alternative solution...

**Comment by Coloryr** [comment]
Do you mean that unsubscribing requires users to manually implement it?

**Comment by Coloryr** [comment]
I think is the control should automatically unsubscribe when released or when the window is closed.

**Comment by Coloryr** [comment]
Of course, providing a method for users to manually release all subscriptions of the control is also feasible, but automatic release within `the framework` is the best choice.

**Comment by Coloryr** [comment]
I have tried unsubscribing in `OnDetachedFromLogicalTreeCore`, but doing so would cause a lot of problems, so I changed to unsubscribe all subscriptions when the window is closed.

**Comment by Coloryr** [comment]
But if it's a single view application(like run in phone), then it will never unsubscribe, but I can't think of a better solution, so now just unsubscribe when `window closed`.

**Comment by Coloryr** [comment]
Or allways use `weakreference`  to store `IObservers` and remove it when need.

**Comment by Coloryr** [comment]
I have found that this problem has become more serious. If animation is enabled, when the window is closed, the property is not unsubscribed, causing the animation to continuously send style data to the control, resulting in the control not actually being released with the release of the window. This also leads to the possibility of memory overflow

**Comment by SuperJMN** [comment]
Any update on this? I mean, does binding to observables directly produce memory leaks?
Thanks :)

**Comment by MrJul** [comment]
There are two issues here:
 - As mentioned above, bindings with observables should have their lifetimes managed by the caller.
 - In v12, animations are now stopped when their visual is detached (#20820).
